src/Type/Php/MethodExistsTypeSpecifyingExtension.php

```diff
@@ -47,15 +46,26 @@ public function specifyTypes(
 		TypeSpecifierContext $context,
 	): SpecifiedTypes
 	{
+		$methodNameType = $scope->getType($node->getArgs()[1]->value);
+		if (!$methodNameType instanceof ConstantStringType) {
+			return new SpecifiedTypes([], []);
+		}
+
 		$objectType = $scope->getType($node->getArgs()[0]->value);
-		if (!$objectType instanceof ObjectType) {
-			if ($objectType->isString()->yes()) {
-				return new SpecifiedTypes([], []);
+		if ($objectType->isString()->yes()) {
+			if ($objectType instanceof ConstantStringType && $objectType->isClassString()) {
```

Comment:
Should we add isClassString on Type as a followup?

Reply:
Of course :)

Reply:
we already have `ConstantStringType->isClassString(): bool`, 

https://github.com/phpstan/phpstan-src/blob/47ba52bd68e13b4242cdace06ea4df7e20159f30/src/Type/Constant/ConstantStringType.php#L75-L84

therefore we need a different name for this new method... any suggestions?

Reply:
isClassStringType

Reply:
And make it nicer for 2.0